.github/workflows/upload_plugin.yml

```
@@ -0,0 +1,83 @@
+name: Upload rpk connect plugin to S3
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      # All runs triggered by tag will really push to S3.
+      # Take care when adding more patterns here.
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+'
+  pull_request:
+    # Keep CI snappy for unrelated PRs
+    paths:
+      - 'resources/plugin_uploader/**'
+      - '.github/workflows/upload_plugin.yml'
+      - '.github/actions/upload_managed_plugin/**'
+      - '.goreleaser.yml'
+  workflow_dispatch: {}
+
+env:
+  # Do dry run in most cases, UNLESS the triggering event was a "tag".
+  DRY_RUN: ${{ ! github.ref_type == 'tag' }}
+
+jobs:
+  upload_rpk_connect_plugin:
+    # Let's make this fast by using a beefy runner.
+    runs-on: ubuntu-latest-32
+    if: ${{ github.repository == 'redpanda-data/connect' }}
```

```suggestion
    if: ${{ github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == 'redpanda-data/connect' }}
```
if i create a PR to this repo from a forked repo `andrewhsu/connect`, the `github.repository` will still have value `redpanda-data/connect`. and the guard will let the control flow into step `Configure AWS credentials` where it will fail. unfortunately, it is poorly [documented](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#github-context) about that specific.

you won't see that situation in this PR because this PR was created from a branch on `redpanda-data/connect` which means you have write permissions to the repo and therefore are allowed to access the `${{ secrets.RP_AWS_CRED_ACCOUNT_ID }}`

---

Appreciate suggestion again.  After doing a bit of research myself, I accept that this somewhat ugly check is the right way!  Updated.